packages/identity/src/identity.ts

```diff
@@ -75,42 +78,35 @@ class ZkIdentity {
     return this.secret
   }
 
-  /**
-   * Creates secret from ZkIdentity
-   * @returns
-   */
-  genSecretFromIdentity() {
-    this.secret = [this.identityNullifier, this.identityTrapdoor]
-  }
 
-  /**
-   * Creates random secret
-   * @param parts number of parts in secret
-   * @returns secret
-   */
-  genRandomSecret(parts = 2) {
-    this.secret = []
-    for (let i = 0; i < parts; i++) {
-      this.secret.push(genRandomNumber())
-    }
+  getSecretHash(): bigint {
+    return poseidonHash(this.secret)
   }
 
+
   /**
-   * Generate commitment from identity secret
-   * @returns identity commitment
+   * Creates secret from ZkIdentity
+   * @returns
    */
-  genIdentityCommitmentFromSecret(): bigint {
-    if (!this.secret.length) throw new Error("Secret is not generated")
-    const secretHash = poseidonHash(this.secret)
-    return poseidonHash([secretHash])
+  genSecret(parts: number = 2): void {
```

Comment:
I guess we always want to make initial secret = trapdor ** nullifier and then make SPAM_THRESHOLD degrees of that initial value?